tests/client-dbus/tests/udev/test_udev.py

```diff
@@ -484,17 +492,30 @@ def test_duplicate_pool_name(
             (luks_tokens, non_luks_tokens) = (
                 [
                     dev
-                    for sublist in (pool_tokens[i] for i in encrypted_indices)
+                    for sublist in (
+                        pool_tokens[i]
+                        for i in (encrypted_indices if _LEGACY_POOL is not None else [])
+                    )
                     for dev in sublist
                 ],
                 [
                     dev
-                    for sublist in (pool_tokens[i] for i in unencrypted_indices)
+                    for sublist in (
+                        pool_tokens[i]
+                        for i in (
+                            unencrypted_indices
+                            if _LEGACY_POOL is not None
+                            else unencrypted_indices + encrypted_indices
+                        )
+                    )
                     for dev in sublist
                 ],
             )
 
-            wait_for_udev(CRYPTO_LUKS_FS_TYPE, self._lb_mgr.device_files(luks_tokens))
+            wait_for_udev(
+                CRYPTO_LUKS_FS_TYPE if _LEGACY_POOL is not None else STRATIS_FS_TYPE,
+                self._lb_mgr.device_files(luks_tokens),
```

Comment:
Change this line to:
```
self._lb_mgr.device_files(luks_tokens if _LEGACY_POOL is not None else non_luks_tokens),
```

and the failure will be resolved.

Also, in the case where ```_LEGACY_POOL``` is  ```None```, the expression will be identical to the one on line 519. But that should be acceptable.

Reply:
My last change seemed to fix the problem! Resolving.